src/elf.rs

```diff
@@ -750,7 +789,11 @@ impl<'a> Elf<'a> {
         header.e_machine = machine.0;
         header.e_type = header::ET_REL;
         header.e_shoff = sh_offset;
-        header.e_shnum = self.nsections;
+        header.e_shnum = if self.nsections >= SHN_LORESERVE.into() {
```

Comment:
was hoping this would have used the need boolean, but it's not too important. going to push a test and merge this :)

Reply:
This is different though. This is whether the number of sections is too large, the boolean is whether the symtab references a large section index. Often these will be the same, but it is not required.

Reply:
What do you mean? `need_symtab_index = self.nsections >= SHN_LORESERVE.into()`

Reply:
Yes, but that test is not exact, and there's a comment above it explaining. There's potential to refactor that in future, and keeping these checks separate (as they logically are) makes it less fragile for refactors.